### PR TITLE
fix(editor): reparent pasted shapes into frame when they land inside one

### LINKED ISSUE #5918: Shape pasted within frame is not parented to the frame
### What happened?

I'm running into a possible bug where some shapes on a canvas that someone else created are within a frame, but aren't part of it. That produces the confusing behavior that any note added to that frame will stay behind those shapes, even if brought to the front. Conversely, sending those shapes to the back will hide them behind the frame.

I was able to fix the problem by dragging the affected shapes ( out and then back in, but the same fix does not work for an arrow shape that is bound on one edge. I suspect that the other shapes an questions previously had arrows bound as well.

### How can we reproduce the bug?

_I haven't found a repro for the box shapes yet. The following only describes the arrow case._

Example: https://www.tldraw.com/f/k_v8j72SFRLuELRsDHqhU?d=v-1030.-949.3320.2338.RKxR33fna4b4I8gaTQQIk

1. Create a frame
2. Add a box to the frame
3. Add an arrow to the frame and connect one end to the box
4. Add a note to the frame and make it overlap the box and arrow. Verify that the arrow is still on top of the box.
5. Bring the note forward. Verify that the arrow is still on top of the box.

Image

### What browsers are you seeing the problem on?

Chrome

### Where did this happen?

The tldraw.com website

### Contact Details

_No response_

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by steveruizok**
Closes #5918

When shapes are pasted (Ctrl+V) and `pasteParentId` resolves to the current page (e.g. nothing is selected, or selected shapes share the page as their common ancestor), shapes are positioned at the viewport center or their original position but no check is made for whether they landed inside a frame. This causes z-ordering confusion: notes added to the frame stay behind the misparented shapes, and sending those shapes to back hides them behind the frame.

This PR adds a post-positioning reparent check in `putContentOntoCurrentPage`. After shapes are positioned, if the paste parent is the page, we compute the center of all pasted root shapes and use `getShapeAtPoint` to find a frame at that center — the same approach already used for paste-at-point. If a frame is found, the root shapes are reparented into it. The pasted shapes themselves are excluded from being considered as parents to prevent frame-into-self reparenting.

https://github.com/user-attachments/assets/a19e2bc2-d04d-42ca-9e20-8bcf89f8fcdf

### Change type

- [x] `bugfix`

### Test plan

1. Create a frame on the canvas
2. Create a shape outside the frame, select it, copy it (Ctrl+C)
3. Deselect all shapes
4. Pan so the viewport is centered on the frame
5. Paste (Ctrl+V) — the shape should become a child of the frame

- [x] Unit tests

### Release notes

- Fixed shapes pasted with Ctrl+V not being parented to a frame when they land inside one.

---

> [!NOTE]
> **Medium Risk**
> Changes paste parenting behavior by auto-reparenting shapes after they’re positioned, which can affect hierarchy/z-order outcomes in edge cases with overlapping frames or groups; covered by new unit tests.
> 
> **Overview**
> Fixes paste behavior so that when content is pasted onto the page (i.e. no explicit paste parent), the editor **post-processes the newly positioned root shapes** and automatically reparents any that ended up inside a frame using `getDroppedShapesToNewParents` + `reparentShapes`.
> 
> Adds unit tests covering reparenting when pasting at viewport center and with `preservePosition`, and updates an ordering expectation in `paste.test.ts` to reflect the new frame-overlap reparenting behavior.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 491d54e3dced83b87256ccde4e466dbeab8900db. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/editor/src/lib/editor/Editor.ts [hunk 1]
```diff
@@ -9601,6 +9602,31 @@ export class Editor extends EventEmitter<TLEventMap> {
...
+				}
+			)
+			reparenting.forEach((childrenToReparent, newParentId) => {
+				if (childrenToReparent.length === 0) return
+				this.reparentShapes(
+					childrenToReparent.map((s) => s.id),
+					newParentId
+				)
+			})
+		}
```
[the remark this answers is not in the document]
Comment (steveruizok) [reply packages/editor/src/lib/editor/Editor.ts]: not accurate
